cpp/proto/substrait/extension_rels.proto

```diff
@@ -58,3 +58,14 @@ message NamedTapRel {
   // If empty, field names will be automatically generated.
   repeated string columns = 3;
 }
+
+message SegmentedAggregateRel {
+  // Grouping keys of the aggregation
+  repeated substrait.Expression.ReferenceSegment grouping_keys = 2;
```

Comment:
From looking at existing substrait messages. `FieldReference` is what is commonly used to represent "keys" so I think we should use that type here. @westonpace WDTY?

Reply:
IIRC, Weston's said this is fine in an extension.

Reply:
@rtpsw What's the issue with using `FeildReference`? here

Reply:
There no issue; using `FieldReference` is an alternative. Because `FieldReference` supports non-direct references, such as a reference to multiple fields, the Acero code accepting it would only support a `ReferenceSegment` under it and would fail on anything else. Since we're dealing with a Substrait extension, there's no requirement to use a `FieldReference` or anything, and we can use the exact kind of reference needed, making things simpler.

Reply:
Fixed.